### PR TITLE
Print if on a single line.

### PR DESCRIPTION
**Pull request by SanderSpies**
Related to #375

Not finished yet. Opening this PR for discussion.

It's not clear to me yet how long constants or idents (e.g "this is a a very long constant string") should be handled.

Also, I think constants and idents are the only ones that I need to check for. Please correct me if I'm wrong in my assumptions.

Other recommendations are of course also welcome.

### CODE REVIEW

File: formatTest/unit_tests/expected_output/if.re [hunk 1]
```diff
@@ -84,13 +70,7 @@ if printIfFirstArgGreater {
...
 };
 
-if (
-  if x {
-    true
-  } else {
-    false
-  }
-) {
+if (if x {true} else {false}) {
```
Comment (kassens) [inline formatTest/unit_tests/expected_output/if.re]: I find the original easier to read, even if it uses up more lines.

File: src/reason_parser_message.ml [hunk 2]
```diff
@@ -10,29 +10,29 @@ let message =
         "<SYNTAX ERROR>\n"
```
Comment (kassens) [inline src/reason_parser_message.ml]: Just sent #570 to get these changes out. I ran into this as well.

File: formatTest/unit_tests/expected_output/if.re (hunk 1 above)
Comment (SanderSpies) [inline formatTest/unit_tests/expected_output/if.re]: I'm pretty neutral on this, besides that an if in an if test is a bit confusing.

**Comment by jberdine** [comment]
I like the short form.  I think that when evaluating choices like this, it is important to consider what the alternatives look like in the context of, perhaps a lot of, other code.  Once surrounded by other code, the value the short forms give by allowing more code to be seen at once is significant.

File: src/reason_parser_message.ml (hunk 2 above)
Comment (yunxing) [inline src/reason_parser_message.ml]: This will gone after rebasing against master.

**Comment by SanderSpies** [comment]
Closed because #621

**Pull request closed without merging**